### ISSUE TITLE
Wrong exponent parsed for Float

### ISSUE THREAD

**Issue by plokhotnyuk** [ngs-doo/dsl-json#122]
The following code prints `0.11596031` instead of `1.1596031`:
```
val bytes = "[1.1596030607005024]".getBytes
val dslJsonScala = new DslJsonScala(new DslJson[Any](new DslJson.Settings[Any]()))
val reader = dslJsonScala.json.newReader()
reader.process(bytes, bytes.length)
reader.read()
val xs = NumberConverter.FLOAT_ARRAY_READER.read(reader)
println(xs(0))
```

**Comment by zapov** [comment]
Tnx for the bug report. Fixed in latest commit: https://github.com/ngs-doo/dsl-json/commit/a6e29586c7671eb9f32b6dd1528471a3251212ca

**Comment by plokhotnyuk** [comment]
What about revising of the testing strategy and adding more generative (property based) tests using wider range of numbers (like double, BigDecimal) instead of fixed cases?

**Comment by zapov** [comment]
There are some, but obviously not enough ;(
There is also extensive testing outside of lib, but not sure if it was reported there.

**Comment by zapov** [comment]
I've released v1.9.1 which has a fix for this problem.
Nice work on finding such problems :tada:

**Comment by plokhotnyuk** [comment]
Nothing special, I just put DSL-JSON parser to [the jsoniter-scala's reader test for floats](https://github.com/plokhotnyuk/jsoniter-scala/blob/c5b26120c5695870923f7fd25e8562e83dc39b49/jsoniter-scala-core/src/test/scala/com/github/plokhotnyuk/jsoniter_scala/core/JsonReaderSpec.scala#L2056-L2084).
